src/main/java/eu/darkbot/api/entities/other/Formation.java

```diff
@@ -70,12 +70,16 @@ public static Formation of(int formationId) {
 
     public static Formation of(String id) {
         for (Formation formation : values())
-            if (formation.id != null && id.endsWith(formation.id))
+            if (formation.endsWith(id))
                 return formation;
 
         return null;
     }
 
+    public boolean endsWith(String formationId) {
```

Comment:
I'd avoid this, it makes little sense, may make the impression that this is a string. if you want to name it `matches(String formationId)` or something else sure, but don't use `endsWith`, how it matches is an implementation detail